File: elixir-to-java/elixir_api_server/lib/elixir_api_server_web/controllers/message_controller.ex
defmodule ElixirApiServerWeb.MessageController do
  use ElixirApiServerWeb, :controller

  @moduledoc false

  def list( conn, _params ) do
    messages = ElixirApiServerWeb.MessageModel.list()
    render( conn, "list.json", api_data: messages )
  end

  def get( conn, %{"id" => id} ) do
    message = ElixirApiServerWeb.MessageModel.get(id)
    render( conn, "message.json", api_data: message )
  end

  def create( conn, %{"message" => message} ) do
    message = ElixirApiServerWeb.MessageModel.create(message)
    render( conn, "create.json", api_data: message )
  end

  def update( conn, %{"id" => id, "message" => message} ) do
    message = ElixirApiServerWeb.MessageModel.update(id, message)
    render( conn, "update.json", api_data: message )
  end

  def delete(conn, %{"id" => id}) do
    ElixirApiServerWeb.MessageModel.delete(id)
    send_resp(conn, :no_content, "")
  end

  def delete_all(conn, _params) do
    ElixirApiServerWeb.MessageModel.delete_all()
    send_resp(conn, :no_content, "")
  end

end

File: micronaut-to-springboot/micronaut-grpc-api/src/main/java/micronaut/grpc/api/MessageController.java
package micronaut.grpc.api;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.*;
import io.micronaut.validation.Validated;
import jakarta.inject.Inject;
import micronaut.grpc.api.client.MessageClient;
import micronaut.grpc.api.model.MessageModel;

import javax.validation.constraints.NotNull;
import java.util.List;

@Validated
@Controller("/message")
public class MessageController {

    private MessageClient client;

    @Inject
    public MessageController(MessageClient client) {
        this.client = client;
    }

    @Get
    public List<MessageModel> list() {
        return client.list();
    }

    @Get("/{id}")
    public MessageModel findById(String id) {
        return client.getById(id);
    }

    @Post(consumes = MediaType.APPLICATION_JSON)
    public MessageModel create(@Body @NotNull MessageModel message) {
        return client.create(message.getMessage());
    }

    @Put(value = "/{id}", consumes = MediaType.APPLICATION_JSON)
    public MessageModel update(String id, @Body @NotNull MessageModel message) {
        return client.update(id, message.getMessage());
    }

    @Delete(value = "/{id}")
    public HttpResponse deleteById(String id) {
        client.deleteById(id);
        return HttpResponse.noContent();
    }

    @Delete
    public HttpResponse deleteAll() {
        client.deleteMessages();
        return HttpResponse.noContent();
    }
}
